phpseclib/Common/Enum.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace phpseclib\Common;
+
+abstract class Enum
+{
+    protected static $constants=null;
+
+    protected static function init()
+    {
+        if (self::$constants==null) {
+            $reflect = new \ReflectionClass(get_called_class());
```

Comment:
The master branch (which this PR targets) only works on PHP 5.6+. Most significantly, every file in File/ASN1/Maps sets a class constant to an array, which wasn't possible until 5.6:

http://php.net/manual/en/migration56.new-features.php

A lot of the files also use constant expressions. The end result being that all of this greatly facilitates re-use.